src/main/java/io/cdap/plugin/repartitioner/Repartitioner.java

```diff
@@ -35,19 +34,17 @@
 public class Repartitioner extends SparkCompute<StructuredRecord, StructuredRecord> {
 
   public static final String PLUGIN_NAME = "Repartitioner";
-  private Config config;
+  private RepartitionerConfig config;
 
-  public Repartitioner(Config config) {
+  public Repartitioner(RepartitionerConfig config) {
     this.config = config;
   }
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
-
-    if (config.getPartitions(1) > 0) {
-      throw new IllegalArgumentException("Number of partitions should be greater than zero.");
-    }
+    FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validate(failureCollector);
```

Comment:
missed validation check

Reply:
The validation check wasn't added based on discussion with Vinisha
https://github.com/data-integrations/topn/pull/12#discussion_r323958674